check_openvpn

```diff
@@ -182,15 +236,22 @@ def main(argv=None):
         return critical('digest creation failed')
 
     if args.tls_auth:
+        # see openvpn source code "src/openvpn/crypto.h", "src/openvpn/crypto.c" and "src/openvpn/crypto_backend.h"
+        # 64 byte cipher direction 0, 64 byte hmac direction 0, 64 byte cipher direction 1, 64 byte hmac direction 1 (=> 2048 bit)
         key = readkey(args.tls_auth)
         if key == None: return critical('cannot read tls auth file')
-        index_start = HMAC_CLIENT_KEY_START * 2
-        index_end = (HMAC_CLIENT_KEY_START + digest_size) * 2
-        key = key[index_start:index_end]
-
-    if key: key = binascii.unhexlify(key)
+        index_start = MAX_CIPHER_KEY_LENGTH + MAX_HMAC_KEY_LENGTH + MAX_CIPHER_KEY_LENGTH
+        index_end = index_start + MAX_HMAC_KEY_LENGTH
+        client_key = key[index_start:index_end]
+        index_start = MAX_CIPHER_KEY_LENGTH
+        index_end = index_start + MAX_HMAC_KEY_LENGTH
+        server_key = key[index_start:index_end]
+        if args.tls_auth_inverse: client_key, server_key = server_key, client_key
+        # reduce key size to required size
+        client_key = client_key[:digest_size]
+        server_key = server_key[:digest_size]
 
-    return checkserver(args.host, args.port, args.tcp, args.timeout, key, digest, args.retrycount)
+    return check(args.host, args.port, args.tcp, args.timeout, digest, digest_size, client_key, server_key, args.retrycount, args.validate)
```

Comment:
This requires that client_key and others are set, but that is not always the case.